### PR TITLE
TR: fix definition of ν

### PR DESCRIPTION
**Pull request by dpo**
The definition in the paper is

$$
\nu_k = 1 / (L(x_k) + \alpha^{-1} \Delta_k^{-1}),
$$

and $L(x_k) = \Vert B_k \Vert$.
There is no $\theta$ involved.

### CODE REVIEW

**Comment by geoffroyleconte** [comment]
Isn't it 

$$
\nu_k \le 1 / (L(x_k) + \alpha^{-1} \Delta_k^{-1})
$$

in the paper?
I've tried to launch the benchmarks tables and I experienced instabilities with unconstrained FH on every random seed I've tried.

**Comment by dpo** [comment]
In principle, there is no reason to take a smaller step size than necessary.

What kind of instabilities?

**Comment by geoffroyleconte** [comment]
Issues with DifferentialEquations.jl

**Comment by rjbaraldi** [comment]
This is probably because FH doesn't actually satisfy our problem assumptions. I recall now that my experience is also that DifferentialEquations.jl fails occasionally.

**Comment by dpo** [comment]
@geoffroyleconte @rjbaraldi What do you think of these numerical results?

File: src/TR_alg.jl [hunk 1]
```diff
@@ -250,7 +249,7 @@ function TR(
       (has_bounds(f) || subsolver == TRDH) ?
       set_bounds!(ψ, max.(-Δk, l_bound - xk), min.(Δk, u_bound - xk)) : set_radius!(ψ, Δk)
     end
-    ν = 1 / (νInv + 1 / (Δk * α))
+    ν = 1 / (λmax + 1 / (Δk * α))
```
Comment (geoffroyleconte) [inline src/TR_alg.jl:252]:
```suggestion
    αΔ = α * Δk
    ν = αΔ / (1 + λmax * (1 +  αΔ))
```
?

**Review by geoffroyleconte — CHANGES_REQUESTED** [review]

File: src/TR_alg.jl (hunk 1 above)
Comment (dpo) [reply src/TR_alg.jl:252]: I don't understand. That's the change I committed already.

Comment (geoffroyleconte) [reply src/TR_alg.jl:252]: You did not change it inside the while loop.

Comment (dpo) [reply src/TR_alg.jl]: Oh duh.

**Comment by dpo** [comment]
@geoffroyleconte What do you think of these demo results?

**Comment by geoffroyleconte** [comment]
I am wondering whether we should use
$$\nu_k = \frac{\alpha \Delta_k}{1 + \Vert B_k \Vert (1 + \alpha \Delta_k)}$$
or
$$\nu_k = \frac{1}{\alpha^{-1} \Delta_k^{-1} +\Vert B_k \Vert (1 + \alpha^{-1} \Delta_k^{-1})}$$
?
I observe different results with these 2 expressions. I would choose the 2nd because we set $\alpha$ to `1 / eps()`, but I'm not 100% sure.
I both cases, this changes the benchmarks of the TRDH paper (I've created a branch on my fork with the current state of this repo to keep the results). Maybe we should also experiment more with the value of $\alpha$ in another PR (the old implementation used the parameter $\theta$ set to $10^{-3}$).

**Comment by dpo** [comment]
> I observe different results with these 2 expressions. I would choose the 2nd because we set $\alpha$
 to `1 / eps()`, but I'm not 100% sure.

I agree that if $\alpha$ is that small, the second expression would be better.

The old and new formulae coincide if $\alpha = \theta^{-1} \Delta^{-1}$. If we assume that $\Delta$ remains $\Theta(1)$, that means $\alpha \approx \theta^{-1}$, i.e., $\alpha \approx 10^3$ in this case.

**Comment by geoffroyleconte** [comment]
> > I observe different results with these 2 expressions. I would choose the 2nd because we set α
> > to `1 / eps()`, but I'm not 100% sure.
> 
> I agree that if α is that small, the second expression would be better.

You mean $\alpha$ large?
> 
> The old and new formulae coincide if α=θ−1Δ−1. If we assume that Δ remains Θ(1), that means α≈θ−1, i.e., α≈103 in this case.

Yes, the old expression is

$$\nu_k = \frac{1}{\alpha^{-1} \Delta_k^{-1} + \Vert B_k \Vert(1 + \theta)}$$

But the first term of the denominator will change as well if we decrease $\alpha$ (maybe it is not really probablematic).

Should I commit the changes I suggested? And may be tune $\alpha$ differently in the benchmarks in another PR?

**Comment by dpo** [comment]
Yes, let's merge this if you think the results look reasonable. I would really like to introduce proper benchmarks in this repo so we can have a clear view of the performance without skimming through the demos.

File: src/TR_alg.jl [hunk 2]
```diff
@@ -132,8 +132,8 @@ function TR(
   Bk = hess_op(f, xk)
 
   λmax = opnorm(Bk)
-  νInv = (1 + θ) * λmax
-  ν = 1 / (νInv + 1 / (Δk * α))
+  αΔ = α * Δk
```
Comment (geoffroyleconte) [inline src/TR_alg.jl:135]:
```suggestion
  α⁻¹Δ⁻¹ = 1 / (α * Δk)
```

File: src/TR_alg.jl [hunk 3]
```diff
@@ -132,8 +132,8 @@ function TR(
...
+  ν = αΔ / (1 + λmax * (1 +  αΔ))
```
Comment (geoffroyleconte) [inline src/TR_alg.jl:136]:
```suggestion
  ν = 1 / (α⁻¹Δ⁻¹ + λmax * (α⁻¹Δ⁻¹ + 1))
```

File: src/TR_alg.jl [hunk 4]
```diff
@@ -250,7 +249,8 @@ function TR(
       (has_bounds(f) || subsolver == TRDH) ?
       set_bounds!(ψ, max.(-Δk, l_bound - xk), min.(Δk, u_bound - xk)) : set_radius!(ψ, Δk)
     end
-    ν = 1 / (νInv + 1 / (Δk * α))
+    αΔ = α * Δk
```
Comment (geoffroyleconte) [inline src/TR_alg.jl:252]:
```suggestion
    α⁻¹Δ⁻¹ = 1 / (α * Δk)
```

File: src/TR_alg.jl [hunk 5]
```diff
@@ -250,7 +249,8 @@ function TR(
...
+    ν = αΔ / (1 + λmax * (1 +  αΔ))
```
Comment (geoffroyleconte) [inline src/TR_alg.jl:253]:
```suggestion
    ν = 1 / (α⁻¹Δ⁻¹ + λmax * (α⁻¹Δ⁻¹ + 1))
```

**Review by geoffroyleconte — CHANGES_REQUESTED** [review]

**Comment by geoffroyleconte** [comment]
I have instabilities with FH, we would need to change the value of $\alpha$ and / or increase $\epsilon$ (but I can do it in another PR if you want).

**Pull request merged**